src/staker.cairo

```diff
@@ -13,8 +13,14 @@ pub trait IStaker<TContractState> {
     // Transfer the approved amount of token from the caller into this contract and delegates it to the given address
     fn stake(ref self: TContractState, delegate: ContractAddress);
 
-    // Withdraws the delegated tokens from the contract to the given recipient address
-    fn withdraw(
+    // Transfer the specified amount of token from the caller into this contract and delegates the voting weight to the specified delegate
+    fn stake_amount(ref self: TContractState, delegate: ContractAddress, amount: u128);
+
+    // Unstakes and withdraws all of the tokens delegated by the sender to the delegate from the contract to the given recipient address
+    fn withdraw(ref self: TContractState, delegate: ContractAddress, recipient: ContractAddress);
+
+    // Unstakes and withdraws the specified amount of tokens delegated by the sender to the delegate from the contract to the given recipient address
+    fn withdraw_amount(
```

Comment:
these improvements to the interface allow you to approve once and then stake for multiple delegates, or move a partial delegation